nuxeo-features/nuxeo-elasticsearch/nuxeo-elasticsearch-core/src/main/java/org/nuxeo/elasticsearch/api/ESHintQueryBuilder.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * (C) Copyright 2019 Nuxeo (http://nuxeo.com/) and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Contributors:
+ *     Salem Aouana
+ */
+
+package org.nuxeo.elasticsearch.api;
+
+import org.elasticsearch.index.query.QueryBuilder;
+import org.nuxeo.ecm.core.query.sql.model.EsHint;
+
+/**
+ * The ES hint query builder provide the ES {@link org.elasticsearch.index.query.QueryBuilder}. It act as a
+ * converter/transformer from ES hint operator of the NXQL query to the ES operator.
```

Comment:
May be something more compact:
Converts an NXQL ES Hint into an elastic {@link org.elasticsearch.index.query.QueryBuilder}.

Reply:
Please avoid abbreviating Elasticsearch as ES in documentation.